File: Composicion-Agregacion/Ejercicio3/Avion.py
# a) Implementa las clases con sus constructores, getters y setters
class Parte:
    def __init__(self, nombre, peso):
        self.nombre = nombre
        self.peso = peso

    def get_nombre(self):
        return self.nombre

    def set_nombre(self, nombre):
        self.nombre = nombre

    def get_peso(self):
        return self.peso

    def set_peso(self, peso):
        self.peso = peso

    def mostrar_info(self):
        print(f"Parte: {self.nombre}, Peso: {self.peso} kg")

class Avion:
    def __init__(self, modelo, fabricante):
        self.modelo = modelo
        self.fabricante = fabricante
        self.partes = []

    def get_modelo(self):
        return self.modelo

    def set_modelo(self, modelo):
        self.modelo = modelo

    def get_fabricante(self):
        return self.fabricante

    def set_fabricante(self, fabricante):
        self.fabricante = fabricante

    def agregar_parte(self, parte):
        self.partes.append(parte)

    def mostrar_avion(self):
        print(f"Modelo del avión: {self.modelo}")
        print(f"Fabricante: {self.fabricante}")
        print("Partes del avión:")
        for parte in self.partes:
            parte.mostrar_info()

# b) Crea un avión y agrega varias partes

avion = Avion("Boeing 737", "Boeing")

parte1 = Parte("Motor", 2000)
parte2 = Parte("Alas", 1500)
parte3 = Parte("Tren de aterrizaje", 800)
parte4 = Parte("Cabina", 1200)

avion.agregar_parte(parte1)
avion.agregar_parte(parte2)
avion.agregar_parte(parte3)
avion.agregar_parte(parte4)

# c) Muestra la información del avión y sus partes

avion.mostrar_avion()

File: Composicion-Agregacion/Ejercicio3/Avion.java
package Ejercicio3;

import java.util.ArrayList;

public class Avion {
    public String modelo;
    public String fabricante;
    public ArrayList<Parte> partes;

    public Avion(String modelo, String fabricante) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.partes = new ArrayList<>();
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getFabricante() {
        return fabricante;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public void agregarParte(Parte parte) {
        partes.add(parte);
    }

    public void mostrarAvion() {
        System.out.println("Modelo del avion: " + modelo);
        System.out.println("Fabricante: " + fabricante);
        System.out.println("Partes del avion:");
        for (Parte parte : partes) {
            parte.mostrarInfo();
        }
    }

    public static void main(String[] args) {
        Avion avion = new Avion("Boeing 737", "Boeing");

        Parte parte1 = new Parte("Motor", 2000);
        Parte parte2 = new Parte("Alas", 1500);
        Parte parte3 = new Parte("Tren de aterrizaje", 800);
        Parte parte4 = new Parte("Cabina", 1200);

        avion.agregarParte(parte1);
        avion.agregarParte(parte2);
        avion.agregarParte(parte3);
        avion.agregarParte(parte4);

        avion.mostrarAvion();
    }

    
}
